packages/components/button/src/button.ts

```diff
@@ -0,0 +1,53 @@
+import { buildProp } from '@element-plus/utils/props'
+
+import type { ExtractPropTypes } from 'vue'
+
+export const buttonType = [
+  'default',
+  'primary',
+  'success',
+  'warning',
+  'info',
+  'danger',
+  'text',
+] as const
+export const buttonSize = ['', 'large', 'medium', 'small', 'mini'] as const
+export const buttonNativeType = ['button', 'submit', 'reset'] as const
+
+export const buttonProps = {
+  type: buildProp({
+    type: String,
+    values: buttonType,
+    default: 'default',
+  } as const),
```

Comment:
Just one question, writing the props this way would seem to increase the mental burden of developers(some of them might not have the context and have no idea why we were using these. It would be better to put a simple comment here IMHO.

Reply:
OK.

Reply:
I read through the `buildProp` method and found it extremely helpful for constraining the types but it could be a little bit difficult for amateur typescript users though. some documentation would have been great. :heart: